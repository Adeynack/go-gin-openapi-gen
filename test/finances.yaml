openapi: 3.0.1

info:
  title: Finances
  version: 1.0.0

paths:

  /books:
    get:
      operationId: books.list
      description: |
        Get a list of all the books to which the current user has access.
        Users with 'admin' role will still the books to which they explicitly√¶
        have access to. Use the 'all=true' query parameter to see all of them.
      parameters:
      - $ref: '#/components/parameters/ListAll'
      responses:
        '200':
          description: List of books
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/BookList'

components:

  parameters:

    ListAll:
      name: list-all
      in: query
      description: |
        Force the complete list to be returned. This will make the operation fail
        if current user does not have necessary rights.
      required: false
      schema:
        type: boolean


  schemas:

    BookId:
      type: integer
      format: int64

    UserId:
      type: integer
      format: int64

    Book:
      type: object
      properties:
        id:
          $ref: '#/components/schemas/BookId'
        name:
          type: string
        owner_id:
          $ref: '#/components/schemas/UserId'

    BookList:
      type: object
      properties:
        items:
          type: array
          items:
            $ref: '#/components/schemas/Book'
