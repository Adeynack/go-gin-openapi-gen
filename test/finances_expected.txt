package api

// Book implements OpenAPI element at #/components/schemas/Book
type Book struct {
	id       *BookId
	name     string
	owner_id *UserId
}

// BookId implements OpenAPI element at #/components/schemas/BookId
type BookId int

// BookList implements OpenAPI element at #/components/schemas/BookList
type BookList struct {
	items []*Book
}

// UserId implements OpenAPI element at #/components/schemas/UserId
type UserId int
